docs/architecture/adr-056-light-client-amnesia-attacks.md

```diff
@@ -0,0 +1,170 @@
+
+An extra dimension to this puzzle is the timeouts. Whilst c1 is relaying votes to its peers and these validators count closer towards the ⅔ threshold needed to send their precommit votes at any moment the timeout could be reached and thus the nodes will precommit nil and ignore any late prevote messages.
+
+This is all to say that such an attack is partly out of the attackers hands. All they can do is tweak the subset of validators that they first choose to gossip the proposal and modify the timings around when they send their prevotes until they reach the desired precondition: n precommits for an earlier proposal and ⅔ precommits for the later proposal. So this is up to the gods of non deterministic behavior to help them out with their plight. I’m not going to allocate the hours to calculate the probability but it could be in the magnitude of 1000’s of blocks trying to get this scenario before the precondition is met.
+
+Obviously, the probability becomes substantially higher as the cabal’s voting power nears ⅔. This is because both n decreases and there is greater tolerance to send prevotes to a greater amount of validators without going overboard and reaching the ⅓ precommit threshold in the first round which would mean they would have to try again.
+
+Once we’ve got our n, we can then forge the remaining signatures for that block (from the f) and bundle them all together and tada we have a forged signed header.
+
+Now we’ve done that, it’s time to find some light clients to fool.
+
+Also critical to this type of attack is that the light client that is connected to our nodes must request a light block at that specific height with which we forged this signed header but this shouldn’t be hard to do. To bring this back to a real context, say our faulty cabal, f, bought some groceries using atoms and then wanted to prove that they did, the grocery owner whips out their phone, runs the light client and f tells them the height they committed the transaction.
+
+An important note here is that because the validator sets are the same between the canonical and the forged block, this attack also works on light clients that verify sequentially. In fact, they are especially vulnerable because they currently don’t run the detector function afterwards.
```

Comment:
does this mean we should run detector for sequential verification too?

Reply:
Yeah I think so, but it's different. I'm not sure if you can perform the attack on intermediate blocks because the validator set is the same but now the last block ID is different so I'm not sure how you can get the same innocent validators to vote for the next forged block in order to sustain the fork. The only way around it is if you can change the validator set completely in this forged block so that you have 2/3+ control of the next validator set. 

If this is not possible then you don't need to worry about the traces you can just compare the last header and if it's incorrect form evidence from that.

Reply:
don't you need 2/3+ malicious nodes to pull this?

Reply:
This, as in an attack on a sequentially validating light client. No theoretically it's possible with 1/3+ although extremely difficult. With 2/3+ it becomes easy

Reply:
https://github.com/tendermint/tendermint/issues/5445